packages/flutter/test/widgets/autocomplete_test.dart

```diff
@@ -647,6 +647,56 @@ void main() {
     );
   });
 
+  testWidgets('support asynchronous options builder', (WidgetTester tester) async {
+    final GlobalKey fieldKey = GlobalKey();
+    final GlobalKey optionsKey = GlobalKey();
+    late FocusNode focusNode;
+    late TextEditingController textEditingController;
+    Iterable<String>? lastOptions;
+    const delay = Duration(milliseconds: 500);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: RawAutocomplete<String>(
+            optionsBuilder: (TextEditingValue textEditingValue) async {
+              return Future.delayed(delay, () => kOptions.where((String option) {
```

Comment:
This looks good.  I was thinking we could have both tests, but I think it's fine like this, this is the most important thing to test.